bundler/lib/bundler/cli/init.rb

```diff
@@ -27,25 +28,41 @@ def run
 
         spec = Bundler.load_gemspec_uncached(gemspec)
 
-        File.open(gemfile, "wb") do |file|
+        File.open(@gemfile, "wb") do |file|
           file << "# Generated from #{gemspec}\n"
           file << spec.to_gemfile
         end
       else
-        File.open(File.expand_path("../templates/#{gemfile}", __dir__), "r") do |template|
-          File.open(gemfile, "wb") do |destination|
+        File.open(File.expand_path(template_path, __dir__), "r") do |template|
+          File.open(@gemfile, "wb") do |destination|
             IO.copy_stream(template, destination)
           end
         end
       end
 
-      puts "Writing new #{gemfile} to #{SharedHelpers.pwd}/#{gemfile}"
+      puts "Writing new #{@gemfile} to #{SharedHelpers.pwd}/#{@gemfile}"
     end
 
     private
 
-    def gemfile
-      @gemfile ||= Bundler.preferred_gemfile_name
+    def preferred_gemfile_name
+      gemfile_name = options[:gemfile] || Bundler.preferred_gemfile_name
+
+      if File.exist?(gemfile_name) && gemfile_name != "Gemfile"
+        Bundler.ui.warn "Bundler will name the gemfile \"Gemfile\" because " \
+                        "#{gemfile_name} already exists at #{File.expand_path(gemfile_name)}. " \
+                        "To silence this warning, run `bundle init --gemfile Gemfile`."
```

Comment:
Sorry for going around in circles with this, but I'm not sure this deprecation makes sense anymore, if we add a CLI flag? Maybe it's better to keep the current behavior of always generating a file named `Gemfile`, regardless of `BUNDLE_GEMFILE`, unless `--gemfile` is given.

Reply:
I like the idea of having only one way of doing things. So it makes sense for me to ignore `BUNDLE_GEMFILE` and only consider `--gemfile`.

Reply:
Alright, in that case I think maybe all the changes in this file can be removed and only change 

```ruby
      @gemfile ||= Bundler.preferred_gemfile_name
```

to

```ruby
      @gemfile ||= options[:gemfile] || Bundler.preferred_gemfile_name
```

Reply:
OK

Reply:
Done.